src/libs/PolicyDistanceRatesUtils.ts

```diff
@@ -22,4 +24,13 @@ function validateRateValue(values: FormOnyxValues<RateValueForm>, currency: stri
     return errors;
 }
 
-export default validateRateValue;
+/**
+ * Get the optimistic rate name in a way that matches BE logic
+ * @param rates
+ */
+function getOptimisticRateName(rates: Record<string, Rate>): string {
+    const existingRatesWithSameName = Object.values(rates).filter((rate) => (rate.name ?? '').startsWith(CONST.CUSTOM_UNITS.DEFAULT_RATE));
```

Comment:
```suggestion
    const existingRatesWithSameName = Object.values(rates ?? {}).filter((rate) => (rate.name ?? '').startsWith(CONST.CUSTOM_UNITS.DEFAULT_RATE));
```
Same reason as other comment